extensions-contrib/kubernetes-overlord-extensions/src/main/java/org/apache/druid/k8s/overlord/KubernetesTaskRunner.java

```diff
@@ -269,17 +260,17 @@ public Optional<InputStream> streamTaskReports(String taskid) throws IOException
   @Override
   public List<Pair<Task, ListenableFuture<TaskStatus>>> restore()
   {
-    List<Pair<Task, ListenableFuture<TaskStatus>>> tasks = new ArrayList<>();
+    List<Pair<Task, ListenableFuture<TaskStatus>>> restoredTasks = new ArrayList<>();
     for (Job job : client.getPeonJobs()) {
       try {
         Task task = adapter.toTask(job);
-        tasks.add(Pair.of(task, joinAsync(task)));
+        restoredTasks.add(Pair.of(task, runOrJoinTask(task, false)));
```

Comment:
I preferred the original separation between `joinTask` and `runTask`. Passing a boolean is cryptic and makes the code less readable.

Reply:
Updated. My thinking was since we added synchronized block (or lock in the previous commit), it's better to have a single place/method to have this complexity. But agree this may make the code less readable.

Reply:
Yeah, that is why I didn't give this feedback originally. But upon reading the code again, it felt cleaner to have them separate.